packages/api/src/module/write/shared/application/events-subscription/events-subscription.ts

```diff
@@ -79,14 +89,19 @@ export class EventsSubscription {
    * See handleEvent for more details how handling event working.
    */
   async start(): Promise<void> {
-    const retryConfig = this.configuration?.retry ?? { retries: 'INFINITELY', backoff: 'EXPONENTIAL', delay: 3000 };
+    const retryConfig = this.configuration.options?.retry ?? {
+      retries: 'INFINITELY',
+      backoff: 'EXPONENTIAL',
+      delay: 3000,
+    };
 
+    // TODO: extract failure recovery to separeate module which will run in bootstrap phase (SRP)
```

Comment:
To follow SRP we can use composition and make something like: 
```.ts
new ResilientEventsSubscription(new CatchupEventsSubscription(new (Volatile / JustListen)EventsSubscription())
```
of course, more abstraction is required and cognitive overhead to understand.

Reply:
What do you mean by 'bug'? That some events may be omitted during catchup? 
Probably not, because I think nothing will be published before application initialization is completed. 
But im pretty sure your solution is better, and queue can handle that doesn't matter application lifecycle, of course :)